src/uproot/behaviors/TBranch.py

```diff
@@ -658,6 +666,12 @@ def to_global(self, global_offset):
         )
 
 
+def _ak_add_doc(array, hasbranches, ak_add_doc):
+    if ak_add_doc and type(array).__module__ == "awkward.highlevel":
```

Comment:
We can't use `isinstance` here because we don't know if `awkward` is installable.

Regarding looseness/strictness: we'll never be able to move `Array` out of `awkward.highlevel`, anyway. That much of the public API is fixed by widespread use.

Also, while we know that `array` is either an `ak.Array` or a `dict`, `list`, or `tuple`, it's nice to narrow in on the three classes in the `awkward.highlevel` submodule, rather than accepting anything that might be defined elsewhere in the Awkward library.